plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationStateUtils.kt

```diff
@@ -9,37 +9,68 @@ import com.intellij.openapi.components.Service
 import com.intellij.openapi.components.State
 import com.intellij.openapi.components.Storage
 import com.intellij.openapi.components.service
+import com.intellij.util.xmlb.Converter
+import com.intellij.util.xmlb.annotations.Attribute
+import com.intellij.util.xmlb.annotations.Property
 import software.aws.toolkits.core.utils.ETagProvider
+import java.time.Duration
+import java.time.Instant
+
+class InstantConverter : Converter<Instant>() {
+    override fun toString(value: Instant): String = value.toEpochMilli().toString()
+
+    override fun fromString(value: String): Instant = Instant.ofEpochMilli(value.toLong())
+}
+
+data class DismissedNotification(
+    @Attribute
+    val id: String = "",
+    @Attribute(converter = InstantConverter::class)
+    val dismissedAt: Instant = Instant.now(),
+)
+
+data class NotificationDismissalConfiguration(
+    @Property
+    var dismissedNotifications: MutableSet<DismissedNotification> = mutableSetOf(),
+)
 
 @Service
 @State(name = "notificationDismissals", storages = [Storage("aws.xml", roamingType = RoamingType.DISABLED)])
 class NotificationDismissalState : PersistentStateComponent<NotificationDismissalConfiguration> {
     private val state = NotificationDismissalConfiguration()
+    private val retentionPeriod = Duration.ofDays(60) // 2 months
 
     override fun getState(): NotificationDismissalConfiguration = state
 
     override fun loadState(state: NotificationDismissalConfiguration) {
-        this.state.dismissedNotificationIds.clear()
-        this.state.dismissedNotificationIds.addAll(state.dismissedNotificationIds)
+        this.state.dismissedNotifications.clear()
```

Comment:
Why do we need to clear the set? Wont each value anyway be unique since it's a set?